core/include/emulate_ops.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2018 Alexandro Sanchez Bach <alexandro@phi.nz>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *   1. Redistributions of source code must retain the above copyright notice,
+ *      this list of conditions and the following disclaimer.
+ *
+ *   2. Redistributions in binary form must reproduce the above copyright
+ *      notice, this list of conditions and the following disclaimer in the
+ *      documentation and/or other materials provided with the distribution.
+ *
+ *   3. Neither the name of the copyright holder nor the names of its
+ *      contributors may be used to endorse or promote products derived from
+ *      this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef HAX_CORE_EMULATE_OPS_H_
+#define HAX_CORE_EMULATE_OPS_H_
+
+#define FASTOP_ALIGN  0x10
+#define FASTOP_OFFSET(size) ( \
+    ((size) == 8) ? (3 * FASTOP_ALIGN) : \
+    ((size) == 4) ? (2 * FASTOP_ALIGN) : \
+    ((size) == 2) ? (1 * FASTOP_ALIGN) : \
+                    (0 * FASTOP_ALIGN))
+
+/* Instruction handlers */
+extern void *em_not;
```

Comment:
nit: So now we are declaring all the handlers as plain pointers instead of function pointers. It's great that it fixes the linker errors, but why? The cost is code readability, although that's not a big issue. I can accept this change, but I just wonder if you have tried other approaches, like decorating the function declaration as you've done with `fastop_dispatch()`:

```
void __cdecl (*em_XXX)();
```

Does that not work for `em_XXX`?

Reply:
Although replacing function pointers with plain `void*` pointers seems like an easy/dirty fix for linking issues, my reasoning was:
1. The `em_XXX` handlers don't respect any calling convention (they are 2-instruction functions with arbitrarily chosen input/output registers).
2. C/C++ code will never call these functions, since they are handled internally by `fastop_dispatch`. The compiler doesn't need to understand what these pointers are.

However, I agree that for the sake of readability (and to make clear we are pointing to *code*), we should replace it with function pointers. I will apply your suggestion in my next commit.

Reply:
I see, that's really thoughtful of you! One last thing I'm curious about: why does decorating `em_XXX` with `__cdel` fix the linking error?

nit: Now that these functions don't really conform to the cdecl calling convention, could you add a short comment above the `typedef`, so people don't find the `__cdecl` part misleading?

Reply:
> why does decorating em_XXX with __cdecl fix the linking error?

OK I think I get it. We add `__cdecl` just for its simple name mangling scheme. The error is due to stdcall name mangling, or more precisely, its use of the `@N` suffix. The `__cdecl` decoration forces C references to these symbols to drop the suffix.